hack/check-images.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source .bingo/variables.env
+
+VALUES=charts/values.global.yaml
+VERSION=$(${YQ} '.version' "$VALUES")
+
+check_image() {
+  IMAGE=$1
+  TAG=$2
+  GMP_VERSIONED=${3:-false}
+
+  LATEST=$(docker run gcr.io/go-containerregistry/crane ls "${IMAGE}" | grep "^v[0-9]" | sort -V | tail -1)
+
+  if [[ $GMP_VERSIONED = true && ! "$TAG" =~ ^v${VERSION}.*$ ]]; then
+    printf "GMP Version is %q, but tag %q of %q does not match\n" "$VERSION" "$TAG" "$IMAGE" && exit 1
+  fi
+
+  if [ "$TAG" != "$LATEST" ]; then
+    printf "%s is %q, latest is %q" "$IMAGE" "$ALERTMANAGER_TAG" "$ALERTMANAGER_LATEST"
```

Comment:
issue: shouldn't this be using `$TAG` and `$LATEST`?

same with the `docker manifest inspect` call below